ThesesFR.js

```diff
@@ -108,53 +107,61 @@ function scrape(doc, url) {
+		if (!title) throw new Error("Reccord must contains a title to be imported");
 
-		const newItem = new Zotero.Item();
+		let newItem = new Zotero.Item();
 		newItem.itemType = 'thesis';
 		newItem.title = title;
 
 		ZU.xpath(xmlDoc, '//marcrel:aut//foaf:Person/foaf:name | //marcrel:dis//foaf:Person/foaf:name', ns).forEach((auth) => {
-			const author = ZU.cleanAuthor(auth.textContent, 'author', true);
+			let author = ZU.cleanAuthor(auth.textContent, 'author', true);
 			newItem.creators.push(author);
 		});
 
 		// Supervisor(s) must be considered as contributor(s) for french thesis
 		ZU.xpath(xmlDoc, '//marcrel:ths//foaf:Person/foaf:name', ns).forEach((sup) => {
-			const supervisor = ZU.cleanAuthor(sup.textContent, 'contributor', true);
+			let supervisor = ZU.cleanAuthor(sup.textContent, 'contributor', true);
 			newItem.creators.push(supervisor);
 		});
 
-		const abstractNotes = ZU.xpath(xmlDoc, '//dcterms:abstract', ns);
-		newItem.abstractNote = abstractNotes.length > 0 ? abstractNotes[0].textContent : undefined;
+		newItem.abstractNote = ZU.xpathText(xmlDoc, '(//dcterms:abstract)[1]', ns);
 
 		// '/s + digit' in url means thesis in preparation
 		newItem.thesisType = url.match(/\/s\d+/) ? 'These en préparation' : 'These de doctorat';
 
-		const organizations = ZU.xpath(xmlDoc, '//marcrel:dgg/foaf:Organization/foaf:name', ns);
-		newItem.university = organizations.length > 0 ? organizations[0].textContent : undefined;
+		newItem.university = ZU.xpathText(xmlDoc, '(//marcrel:dgg/foaf:Organization/foaf:name)[1]', ns);
 
-		const fullDate = ZU.xpathText(xmlDoc, '//dcterms:dateAccepted', ns);
-		const year = ZU.xpathText(xmlDoc, '//dc:date', ns);
+		let fullDate = ZU.xpathText(xmlDoc, '//dcterms:dateAccepted', ns);
+		let year = ZU.xpathText(xmlDoc, '//dc:date', ns);
 
 		// Some old records doesn't have a full date instead we can use the defense year
 		newItem.date = fullDate ? fullDate : year;
-
-		const permalink = ZU.xpath(xmlDoc, '//bibo:Document', ns);
-		newItem.url = permalink.length > 0 ? permalink[0].getAttribute('rdf:about') : url;
+		newItem.url = url;
 		newItem.libraryCatalog = 'theses.fr';
-		newItem.rights = 'Les données de Theses.fr sont sous licence Etalab';
+		newItem.rights = 'Licence Etalab';
+
+		// Keep extra information such as laboratory, graduate schools, etc. in a note for thesis not yet defended
+		let notePrepa = Array.from(doc.getElementsByClassName('donnees-ombreprepa2')).map((description) => {
+			return Array.from(description.getElementsByTagName('p')).map(description => description.textContent.replace(/\n/g, ' ').trim());
+		}).join(' ');
 
-		// Keep extra information such as laboratory, graduate schools, etc.
-		newItem.extra = Array.from(doc.getElementsByClassName('donnees-ombre')).map((description) => {
-			return Array.from(description.getElementsByTagName('p')).map(description => description.textContent.replace(/\n/g, ' '));
+		if (notePrepa) {
+			newItem.notes.push({ description: notePrepa });
```

Comment:
This should be `{note: notePrepa}` as per https://www.zotero.org/support/dev/translators/coding#notes 
Same below